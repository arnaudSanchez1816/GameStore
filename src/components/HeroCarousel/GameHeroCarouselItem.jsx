import style from "./GameHeroCarouselItem.module.css"

function GameHeroCarouselItem({ data }) {
    const { name, image } = data

    return (
        <a
            className={style["hero-carousel-item"]}
            href={image}
            target="_blank"
            rel="noopener noreferrer"
        >
            <img className={style["image"]} src={image} alt={name} />
        </a>
    )
}

export default GameHeroCarouselItem
