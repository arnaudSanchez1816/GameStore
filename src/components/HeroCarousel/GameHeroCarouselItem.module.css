.hero-carousel-item {
    &:visited {
        color: inherit;
    }

    position: relative;
    display: block;
    width: 100%;
    height: 100%;
}

.image {
    border-radius: var(--border-radius-medium);
    display: block;
    object-fit: cover;
    width: 100%;
    height: 100%;
}
