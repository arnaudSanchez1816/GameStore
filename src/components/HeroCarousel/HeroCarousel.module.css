.hero-carousel {
    width: 100%;
    position: relative;
}

.row-container {
    position: relative;
    width: 100%;
    height: 600px;
    display: grid;
    grid-template-columns: 40px 1fr 40px;
    grid-template-areas: "left carousel right";
    gap: 1rem;
}

.row-list {
    position: relative;
    list-style: none;
    width: 100%;
    height: 100%;
    margin: 0;
    padding: 0;
    grid-area: carousel;
}

.row-item {
    position: absolute;
    overflow: hidden;
    flex: 0 0 100%;
    width: 100%;
    height: 100%;
    margin: 0;
    display: block;
    will-change: opacity;
    transition: 400ms opacity;
    opacity: 0;
    pointer-events: none;
}

.row-item:global(.active) {
    opacity: 1;
    pointer-events: auto;
}

.dots-list {
    list-style: none;
    margin: 1rem 0 0;
    padding: 0;
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 0.5rem;
}

.dot {
    position: relative;
    border-radius: 50%;
    width: 1rem;
    height: 1rem;
    padding: 0;
    display: block;
    border: 3px solid #5a5a5a;
    background-color: #5a5a5a;

    &.active {
        background-color: white;
        border-color: transparent;
    }

    &:hover {
        border-color: white;
    }
}

.dot-fill {
    position: absolute;
    inset: -6px;
    border: 3px solid green;
    border-radius: 50%;
}

@media (width<= 800px) {
    .row-container {
        height: 300px;
        overflow: scroll;
        display: block;
    }

    .row-list {
        display: flex;
        flex-flow: row nowrap;
        padding: 0 0.5rem 1rem;
        gap: 0.5rem;
    }

    .row-item {
        position: static;
        opacity: 1;
        pointer-events: auto;
        flex: 1 0 85%;
    }

    .dots-list {
        display: none;
    }
}
