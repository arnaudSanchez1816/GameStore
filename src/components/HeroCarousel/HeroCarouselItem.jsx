import style from "./HeroCarouselItem.module.css"

function HeroCarouselItem({ data }) {
    const { gameId, name, imgLink, price } = data

    return (
        <a className={style["hero-carousel-item"]} href="#">
            <img className={style["image"]} src={imgLink} alt={name} />
            <div className={style["caption"]}>
                <div className={style["caption-container"]}>
                    <p className={style["title"]}>{name}</p>
                    <div className={style["pricing"]}>
                        <span className={style["price"]}>{price}</span>
                        <button className={style["add-to-cart"]} type="button">
                            Add to cart
                        </button>
                    </div>
                </div>
            </div>
        </a>
    )
}

export default HeroCarouselItem
