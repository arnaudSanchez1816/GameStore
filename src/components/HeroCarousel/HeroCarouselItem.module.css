.hero-carousel-item {
    &:visited {
        color: inherit;
    }

    position: relative;
    display: block;
    width: 100%;
    height: 100%;
}

.hero-carousel-item::before {
    content: "";
    position: absolute;
    inset: 0;
    background: linear-gradient(to top, rgb(0 0 0 / 100%), transparent 35%);
}

.image {
    display: block;
    object-fit: cover;
    width: 100%;
    height: 100%;
}

.caption {
    position: absolute;
    inset: 0;
}

.caption-container {
    width: 100%;
    height: 100%;
    padding: 1rem 2rem;
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-template-rows: 1fr auto;
    grid-template-areas:
        ". ."
        "title pricing";
}

.title {
    margin: 0;
    font-size: 1.25rem;
    text-align: left;
    grid-area: title;
}

.pricing {
    display: flex;
    flex-flow: row;
    justify-content: flex-end;
    align-items: flex-end;
    grid-area: pricing;
}

.price {
    font-size: 1.5rem;
    margin-right: 1rem;
    font-weight: bold;
}

.add-to-cart {
    background-color: green;
    border: none;
    border-radius: 20px;

    &:hover {
        background-color: greenyellow;
    }
}
