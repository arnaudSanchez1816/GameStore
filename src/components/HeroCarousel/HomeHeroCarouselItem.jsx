import { Link } from "react-router-dom"
import style from "./HomeHeroCarouselItem.module.css"
import AddToCartButton from "../AddToCartButton/AddToCartButton"

function HomeHeroCarouselItem({ data }) {
    const { name, background_image, price, id } = data

    return (
        <div className={style["hero-carousel-item"]}>
            <Link className={style["game-link"]} to={`/game/${id}`}>
                <img
                    className={style["image"]}
                    src={background_image}
                    alt={name}
                />
            </Link>
            <div className={style["caption"]}>
                <div className={style["caption-container"]}>
                    <p className={style["title"]}>{name}</p>
                    <div className={style["pricing"]}>
                        <span className={style["price"]}>{`${price}â‚¬`}</span>
                        <div className={style["add-to-cart"]}>
                            <AddToCartButton gameId={id} />
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}

export default HomeHeroCarouselItem
