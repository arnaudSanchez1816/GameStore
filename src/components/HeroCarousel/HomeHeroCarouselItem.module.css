.hero-carousel-item {
    position: relative;
    height: 100%;
}

.game-link {
    display: block;
    color: var(--text-color-light);
    width: 100%;
    height: 100%;

    &:visited {
        color: inherit;
    }

    &:hover {
        color: inherit;
    }
}

.game-link::before {
    content: "";
    position: absolute;
    inset: 0;
    background: linear-gradient(to top, rgb(0 0 0 / 100%), transparent 35%);
    border-bottom-left-radius: var(--border-radius-medium);
    border-bottom-right-radius: var(--border-radius-medium);
}

.image {
    border-radius: var(--border-radius-medium);
    display: block;
    object-fit: cover;
    width: 100%;
    height: 100%;
}

.caption {
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    pointer-events: none;
}

.caption-container {
    width: 100%;
    height: 100%;
    padding: 2rem;
    display: flex;
    flex-flow: row nowrap;
    justify-content: space-between;
}

.title {
    margin: 0;
    font-size: 1.25rem;
    text-align: left;
    grid-area: title;
    align-self: flex-end;
}

.pricing {
    display: flex;
    justify-content: flex-end;
    align-items: flex-end;
    grid-area: pricing;
}

.price {
    font-size: 1.5rem;
    margin-right: 1rem;
    font-weight: bold;
}

.add-to-cart {
    flex-shrink: 0;
}

:global(.active) .add-to-cart {
    pointer-events: auto;
}

@media (width<=800px) {
    .caption-container {
        padding: 1rem;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr auto;
        grid-template-areas:
            "title"
            "pricing";
        gap: 1rem;
    }

    .title {
        font-size: 1rem;
        font-weight: bold;
    }

    .price {
        margin-right: 0;
        font-size: 1rem;
    }

    .add-to-cart {
        display: none;
    }

    .game-link::before {
        background: linear-gradient(
            to top,
            rgb(0 0 0 / 100%),
            rgb(0 0 0 / 80%) 30%,
            transparent 60%
        );
    }
}
