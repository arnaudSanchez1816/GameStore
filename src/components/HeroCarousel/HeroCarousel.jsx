import style from "./HeroCarousel.module.css"
import HeroCarouselItem from "./HeroCarouselItem"
import { useEffect, useRef, useState } from "react"
import clsx from "clsx"
import { Icon } from "@iconify-icon/react/dist/iconify.mjs"

const DEFAULT_ITEMS = [
    {
        gameId: 100,
        name: "The Elder Scrolls IV: Oblivion Remastered",
        imgLink:
            "https://images.ctfassets.net/rporu91m20dc/1wGximc864FDh0hPe5mbjI/9039ff982fbf12911290467dcc42d36c/OR_LargeHero_AvailableNow.png?q=70&fm=png",
        price: "50.00€",
    },
    {
        gameId: 110,
        name: "Kingdom Come : Deliverance II",
        imgLink:
            "https://pbs.twimg.com/media/GWeeYlBXwAAipbn?format=jpg&name=4096x4096",
        price: "29.99€",
    },
    {
        gameId: 120,
        name: "Black Myth: Wukong",
        imgLink:
            "https://cdn1.epicgames.com/spt-assets/ca9ef1bcf2f54043baac351366aec677/black-myth-wukong-jz9yr.jpg",
        price: "50.00€",
    },
    {
        gameId: 130,
        name: "Age of Mythology : Retold",
        imgLink:
            "https://image.api.playstation.com/vulcan/ap/rnd/202501/2218/781d28bc15aff7c19074e087cc006f2637c7f6df85f1a000.jpg",
        price: "49.99€",
    },
    {
        gameId: 140,
        name: "Anno 1800 : Gold Edition",
        imgLink:
            "https://cdn2.unrealengine.com/egst-edition-thumbnail-1920x1080-1920x1080-5a00fb27a68e.jpg",
        price: "14.99€",
    },
    {
        gameId: 150,
        name: "Factorio",
        imgLink: "https://cdn.factorio.com/assets/img/artwork/0.17-stable.png",
        price: "29.99€",
    },
]

const dotFillAnim = [
    {
        clipPath: "polygon(50% 0, 50% 0, 50% 0, 50% 0, 50% 0, 50% 0, 50% 50%)",
        easing: "linear",
    },
    {
        clipPath:
            "polygon(50% 0, 100% 0, 100% 0, 100% 0, 100% 0, 100% 0, 50% 50%)",
        easing: "linear",
    },
    {
        clipPath:
            "polygon(50% 0, 100% 0, 100% 100%, 100% 100%, 100% 100%, 100% 100%, 50% 50%)",
        easing: "linear",
    },
    {
        clipPath:
            "polygon(50% 0, 100% 0, 100% 100%, 0% 100%, 0% 100%, 0% 100%, 50% 50%)",
        easing: "linear",
    },
    {
        clipPath:
            "polygon(50% 0, 100% 0, 100% 100%, 0% 100%, 0 0, 0 0, 50% 50%)",
        easing: "linear",
    },
    {
        clipPath:
            "polygon(50% 0, 100% 0, 100% 100%, 0% 100%, 0 0, 50% 0, 50% 50%)",
        easing: "linear",
    },
]

function HeroCarouselNavDot({
    index,
    isActive,
    onClick,
    delayDuration = null,
}) {
    const onDotClick = () => onClick(index)
    const dotFillRef = useRef(null)
    const hasDelay = isActive && delayDuration != null

    useEffect(() => {
        let anim = null
        if (dotFillRef.current) {
            anim = dotFillRef.current.animate(dotFillAnim, {
                easing: "linear",
                duration: delayDuration,
            })
        }

        return () => {
            anim?.cancel()
        }
    }, [isActive, delayDuration])

    return (
        <button
            type="button"
            className={clsx(style["dot"], isActive && style["active"])}
            onClick={onDotClick}
        >
            {hasDelay && <div className={style["dot-fill"]} ref={dotFillRef} />}
        </button>
    )
}

const HERO_CAROUSEL_DEFAULT = {
    autoScroll: false,
    scrollDelay: 5000,
}

function HeroCarousel({
    items = DEFAULT_ITEMS,
    options = HERO_CAROUSEL_DEFAULT,
}) {
    options = { ...HERO_CAROUSEL_DEFAULT, ...options }
    const { autoScroll, scrollDelay } = options
    const [index, setIndex] = useState(0)

    useEffect(() => {
        let interval = null
        if (autoScroll) {
            interval = setInterval(() => {
                let newIndex = index + 1
                if (newIndex >= items.length) {
                    newIndex = 0
                }
                setIndex(newIndex)
            }, scrollDelay)
        }

        return () => {
            if (interval) {
                clearInterval(interval)
            }
        }
    }, [autoScroll, scrollDelay, index, items])

    const onLeftControls = () => {
        setIndex(Math.max(0, index - 1))
    }

    const onRightControls = () => {
        setIndex(Math.min(items.length - 1, index + 1))
    }

    const onNavDotClick = (i) => {
        setIndex(i)
    }

    const dots = []
    for (let i = 0; i < items.length; i++) {
        dots.push(
            <HeroCarouselNavDot
                index={i}
                isActive={index === i}
                onClick={onNavDotClick}
                delayDuration={options.autoScroll ? options.scrollDelay : null}
            />
        )
    }

    return (
        <div className={style["hero-carousel"]}>
            <div className={style["row-container"]}>
                <ol className={style["row-list"]}>
                    {items.map((item, mapIndex) => {
                        const isActive = index === mapIndex

                        return (
                            <li
                                className={clsx(
                                    style["row-item"],
                                    isActive && style["active"]
                                )}
                                key={item.gameId}
                            >
                                <HeroCarouselItem data={item} />
                            </li>
                        )
                    })}
                </ol>
                {index > 0 && (
                    <div className={style["controls-left"]}>
                        <button
                            className={style["controls-btn"]}
                            type="button"
                            onClick={onLeftControls}
                        >
                            <Icon icon="mdi:chevron-left" />
                        </button>
                    </div>
                )}
                {index < items.length - 1 && (
                    <div className={style["controls-right"]}>
                        <button
                            className={style["controls-btn"]}
                            type="button"
                            onClick={onRightControls}
                        >
                            <Icon icon="mdi:chevron-right" />
                        </button>
                    </div>
                )}
            </div>
            <ol className={style["dots-list"]}>
                {dots.map((dot, mapIndex) => (
                    <li key={mapIndex}>{dot}</li>
                ))}
            </ol>
        </div>
    )
}

export default HeroCarousel
