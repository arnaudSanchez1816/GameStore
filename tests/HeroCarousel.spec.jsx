import { test, expect, describe, vi, beforeAll, afterAll } from "vitest"
import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import HeroCarousel from "../src/components/HeroCarousel/HeroCarousel"
import { act } from "react"

describe("Hero Carousel", () => {
    // Mock native animate function
    let originalAnimateFunction
    beforeAll(() => {
        originalAnimateFunction = HTMLDivElement.prototype.animate

        const obj = {
            onfinish: () => {},
            cancel: () => {},
        }

        const animationFunction = function () {
            Promise.resolve().then(async () => {
                act(() => obj.onfinish())
            })

            return obj
        }

        HTMLDivElement.prototype.animate = animationFunction
    })

    afterAll(() => {
        HTMLDivElement.prototype.animate = originalAnimateFunction
    })

    test("to render items", () => {
        const { container } = render(
            <HeroCarousel>
                {[...Array(10)].map((item, index) => (
                    <div key={index}>Game {index}</div>
                ))}
            </HeroCarousel>
        )

        expect(container).toMatchSnapshot()
    })

    test("Only 1 active item", async () => {
        const toRender = (
            <HeroCarousel options={{ autoScroll: false }}>
                {[...Array(10)].map((item, index) => (
                    <div>Game {index}</div>
                ))}
            </HeroCarousel>
        )
        const { container } = render(toRender)

        const gameItems = screen.getAllByText("Game", { exact: false })
        expect(
            gameItems.filter((item) =>
                item.parentElement.classList.contains("active")
            ).length
        ).toBe(1)
    })

    test("the active item change after 500 ms", async () => {
        const toRender = (
            <HeroCarousel options={{ autoScroll: true, scrollDelay: 500 }}>
                {[...Array(10)].map((item, index) => (
                    <div>Game {index}</div>
                ))}
            </HeroCarousel>
        )
        const { container } = render(toRender)

        expect(
            screen
                .getByText("Game 0")
                .parentElement.classList.contains("active")
        ).toBe(true)
        expect(
            screen
                .getByText("Game 1")
                .parentElement.classList.contains("active")
        ).toBe(false)

        await act(async () => {
            await new Promise((res) => {
                setTimeout(res, 600)
            })
        })
        expect(
            screen
                .getByText("Game 0")
                .parentElement.classList.contains("active")
        ).toBe(false)
        expect(
            screen
                .getByText("Game 1")
                .parentElement.classList.contains("active")
        ).toBe(true)
    })

    test("the active to not change", async () => {
        const toRender = (
            <HeroCarousel options={{ autoScroll: false, scrollDelay: 500 }}>
                {[...Array(10)].map((item, index) => (
                    <div>Game {index}</div>
                ))}
            </HeroCarousel>
        )
        const { container } = render(toRender)

        expect(
            screen
                .getByText("Game 0")
                .parentElement.classList.contains("active")
        ).toBe(true)
        expect(
            screen
                .getByText("Game 1")
                .parentElement.classList.contains("active")
        ).toBe(false)

        await act(async () => {
            await new Promise((res) => {
                setTimeout(res, 600)
            })
        })
        expect(
            screen
                .getByText("Game 0")
                .parentElement.classList.contains("active")
        ).toBe(true)
        expect(
            screen
                .getByText("Game 1")
                .parentElement.classList.contains("active")
        ).toBe(false)
    })

    test("Right/Left buttons click to change the active item", async () => {
        const user = userEvent.setup()
        const toRender = (
            <HeroCarousel options={{ autoScroll: false }}>
                {[...Array(10)].map((item, index) => (
                    <div>Game {index}</div>
                ))}
            </HeroCarousel>
        )
        const { container } = render(toRender)

        const game0Item = screen.getByText("Game 0")
        const game1Item = screen.getByText("Game 1")

        expect(game0Item.parentElement.classList.contains("active")).toBe(true)
        expect(game1Item.parentElement.classList.contains("active")).toBe(false)

        const rightButton = screen.getByRole("button", { name: "Right" })
        await user.click(rightButton)

        expect(game0Item.parentElement.classList.contains("active")).toBe(false)
        expect(game1Item.parentElement.classList.contains("active")).toBe(true)

        const leftButton = screen.getByRole("button", { name: "Left" })
        await user.click(leftButton)

        expect(game0Item.parentElement.classList.contains("active")).toBe(true)
        expect(game1Item.parentElement.classList.contains("active")).toBe(false)
    })

    test("Nav dots buttons click to change the active item", async () => {
        const user = userEvent.setup()
        const toRender = (
            <HeroCarousel options={{ autoScroll: false }}>
                {[...Array(10)].map((item, index) => (
                    <div>Game {index + 1}</div>
                ))}
            </HeroCarousel>
        )
        const { container } = render(toRender)

        const game1Item = screen.getByText("Game 1")
        const game10Item = screen.getByText("Game 10")

        expect(game1Item.parentElement.classList.contains("active")).toBe(true)
        expect(game10Item.parentElement.classList.contains("active")).toBe(
            false
        )

        const item10NavDot = screen.getByRole("button", { name: "Item 10" })
        await user.click(item10NavDot)

        expect(game1Item.parentElement.classList.contains("active")).toBe(false)
        expect(game10Item.parentElement.classList.contains("active")).toBe(true)

        const item1NavDot = screen.getByRole("button", { name: "Item 1" })
        await user.click(item1NavDot)

        expect(game1Item.parentElement.classList.contains("active")).toBe(true)
        expect(game10Item.parentElement.classList.contains("active")).toBe(
            false
        )
    })
})
