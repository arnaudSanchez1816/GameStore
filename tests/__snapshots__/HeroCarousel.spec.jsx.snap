// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hero Carousel > to render items 1`] = `
<div>
  <div
    class="_hero-carousel_d37b2b"
  >
    <div
      class="_row-container_d37b2b"
    >
      <ol
        class="_row-list_d37b2b"
      >
        <li
          class="_row-item_d37b2b active"
        >
          <div>
            Game 
            0
          </div>
        </li>
        <li
          class="_row-item_d37b2b"
        >
          <div>
            Game 
            1
          </div>
        </li>
        <li
          class="_row-item_d37b2b"
        >
          <div>
            Game 
            2
          </div>
        </li>
        <li
          class="_row-item_d37b2b"
        >
          <div>
            Game 
            3
          </div>
        </li>
        <li
          class="_row-item_d37b2b"
        >
          <div>
            Game 
            4
          </div>
        </li>
        <li
          class="_row-item_d37b2b"
        >
          <div>
            Game 
            5
          </div>
        </li>
        <li
          class="_row-item_d37b2b"
        >
          <div>
            Game 
            6
          </div>
        </li>
        <li
          class="_row-item_d37b2b"
        >
          <div>
            Game 
            7
          </div>
        </li>
        <li
          class="_row-item_d37b2b"
        >
          <div>
            Game 
            8
          </div>
        </li>
        <li
          class="_row-item_d37b2b"
        >
          <div>
            Game 
            9
          </div>
        </li>
      </ol>
      <div
        class="_controls-right_6c2e0f"
      >
        <button
          aria-label="Right"
          class="_controls-btn_6c2e0f"
          type="button"
        >
          <iconify-icon
            icon="mdi:chevron-right"
          />
        </button>
      </div>
    </div>
    <ol
      class="_dots-list_d37b2b"
    >
      <li>
        <button
          aria-label="Item 1"
          class="_dot_d37b2b _active_d37b2b"
          type="button"
        />
      </li>
      <li>
        <button
          aria-label="Item 2"
          class="_dot_d37b2b"
          type="button"
        />
      </li>
      <li>
        <button
          aria-label="Item 3"
          class="_dot_d37b2b"
          type="button"
        />
      </li>
      <li>
        <button
          aria-label="Item 4"
          class="_dot_d37b2b"
          type="button"
        />
      </li>
      <li>
        <button
          aria-label="Item 5"
          class="_dot_d37b2b"
          type="button"
        />
      </li>
      <li>
        <button
          aria-label="Item 6"
          class="_dot_d37b2b"
          type="button"
        />
      </li>
      <li>
        <button
          aria-label="Item 7"
          class="_dot_d37b2b"
          type="button"
        />
      </li>
      <li>
        <button
          aria-label="Item 8"
          class="_dot_d37b2b"
          type="button"
        />
      </li>
      <li>
        <button
          aria-label="Item 9"
          class="_dot_d37b2b"
          type="button"
        />
      </li>
      <li>
        <button
          aria-label="Item 10"
          class="_dot_d37b2b"
          type="button"
        />
      </li>
    </ol>
  </div>
</div>
`;
